File: src/net/eduard/api/server/CashSystem.java
package net.eduard.api.server;

import net.eduard.api.lib.modules.FakePlayer;

public interface CashSystem {

	public void addCash(FakePlayer player, double amount);

	public void removeCash(FakePlayer player, double amount);

	public double getCash(FakePlayer player);

	public void setCash(FakePlayer player, double amount);

	public default boolean hasCash(FakePlayer player, double amount) {
		return getCash(player) >= amount;
	}

}

File: src/main/java/net/eduard/api/server/CashSystem.kt
package net.eduard.api.server

import net.eduard.api.lib.modules.FakePlayer

interface CashSystem : PluginSystem {

    fun addCash(player: FakePlayer, amount: Double)

    fun removeCash(player: FakePlayer, amount: Double)

    fun getCash(player: FakePlayer): Double

    fun setCash(player: FakePlayer, amount: Double)

    fun hasCash(player: FakePlayer, amount: Double): Boolean {
        return getCash(player) >= amount
    }

}
